### PR TITLE
Add recommended Kubernetes app labels to Crossplane components

### PR DESCRIPTION
**Pull request by nlamirault**
### Description of your changes

- Refactoring metadata labels and selector labels
- Support [Kubernetes recommanded labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/)

Fixes #

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by nlamirault** [comment]
`reviewable` does not seems to be an available target. 

```
> make reviewable
make: *** No rule to make target 'reviewable'.  Stop.
```

**Comment by jbw976** [comment]
Has the build submodule been synced into your local env? Give `make submodules` a try, then try `make reviewable` again.

File: cluster/charts/crossplane/templates/_helpers.tpl [hunk 1]
```diff
@@ -2,13 +2,38 @@
...
-{{- define "chart" -}}
+{{- define "crossplane.chart" -}}
 {{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
+
+{{/*
+Generate basic labels
+*/}}
+{{- define "crossplane.labels" }}
+helm.sh/chart: {{ include "crossplane.chart" . }}
```
[the remark this answers is not in the document]
Comment (nlamirault) [reply cluster/charts/crossplane/templates/_helpers.tpl:20]: @negz I don't think. When you use the `create` command from Helm, the label is created by Helm into the `_helper.tpl` file.

**Comment by nlamirault** [comment]
the output seems ok : 

```
❯ make reviewable
07:57:19 [ .. ] verify dependencies have expected content
all modules verified
07:57:22 [ OK ] go modules dependencies verified
07:57:22 [ .. ] go generate linux_amd64
07:57:48 [ OK ] go generate linux_amd64
07:57:48 [ .. ] go mod tidy
07:57:49 [ OK ] go mod tidy
07:57:49 [ .. ] cleaning generated CRDs
07:57:49 [ OK ] cleaned generated CRDs
07:57:49 [ .. ] Adding all CRDs to Kustomize file for local development
07:57:49 [ OK ] All CRDs added to Kustomize file for local development
07:57:49 [ .. ] Generating install documentation from Helm chart
07:57:49 [ OK ] Successfully generated install documentation
==> Linting /home/nicolas/Projects/Forks/crossplane/cluster/charts/crossplane

1 chart(s) linted, 0 chart(s) failed
07:57:50 [ .. ] verify dependencies have expected content
all modules verified
07:57:54 [ OK ] go modules dependencies verified
07:57:54 [ .. ] golangci-lint
07:57:56 [ OK ] golangci-lint
07:57:57 [ .. ] verify dependencies have expected content
all modules verified
07:58:00 [ OK ] go modules dependencies verified
07:58:01 [ .. ] go test unit-tests
go test: -i flag is deprecated
=== RUN   TestBuild
=== RUN   TestBuild/SuccessfulWithName
=== RUN   TestBuild/ErrNoNameNoMeta
--- PASS: TestBuild (0.00s)
....
PASS
coverage: 38.0% of statements
ok      github.com/crossplane/crossplane/apis/pkg/meta/v1alpha1 0.010s  coverage: 38.0% of statements
?       github.com/crossplane/crossplane/apis/pkg/v1    [no test files]
?       github.com/crossplane/crossplane/apis/pkg/v1alpha1      [no test files]
?       github.com/crossplane/crossplane/apis/pkg/v1beta1       [no test files]
07:58:28 [ OK ] go test unit-tests
```

File: cluster/charts/crossplane/templates/_helpers.tpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (negz) [reply cluster/charts/crossplane/templates/_helpers.tpl:20]: Thank you!

**Review by negz — APPROVED** [review]
This LGTM!

**Comment by dwerder** [comment]
Checked with helm and helm unittest. 
This LGTM!

Only thing I found was, that there seems to be a race condition when you run `helm uninstall` with rbac roles. rbac manager recreates some roles while the uninstall process ist running. But this has nothing todo with this PR.

**Pull request merged**